Source/Particles/Resampling/Resampling.H

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2019-2020 Neil Zaim
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef WARPX_RESAMPLING_H_
+#define WARPX_RESAMPLING_H_
+
+#include "ResamplingTrigger.H"
+
+class WarpXParIter; // forward declaration
+
+/**
+ * \brief An empty base class from which specific resampling algorithms are derived.
+ */
+struct ResamplingAlgorithm
+{
+    virtual void run(WarpXParIter& /*pti*/) {};
```

Comment:
Since this class is essentially an interface, you could make it an abstract class:

```suggestion
    virtual void run(WarpXParIter& /*pti*/) {} = 0;
    virtual ~ResamplingAlgorithm () =  default;
```

Reply:
Good idea, just omit the body (little typo):
```suggestion
    virtual void run(WarpXParIter& /*pti*/) = 0;
```

Reply:
you can also implement `run()` as an `void operator()(WarpXParIter& /*pti*/)` to make it a functor - an object that you can call like a function `(...)` (instead of `.run(...)`).